modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts

```diff
@@ -0,0 +1,185 @@
+            AlloyEvents.run(NativeEvents.focusout(), (_comp, se) => {
+              spec.onAction(Value.get(se.event.target));
+            }),
+            AlloyEvents.run(NativeEvents.change(), (_comp, se) => {
+              spec.onAction(Value.get(se.event.target));
+            })
+          ]),
+          Keying.config({
+            mode: 'special',
+            onEnter: (comp) => {
+              spec.onAction(Representing.getValue(comp));
+              return Optional.some(true);
+            },
+            onEscape: goToParent,
+            onUp: (_comp) => {
+              increase(false);
+              // TOFIX: now it preserve the focus but it put the selection at the end of the input
+              return Optional.some(true);
+            },
+            onDown: (_comp) => {
+              decrease(false);
+              return Optional.some(true);
+            }
+          })
+        ])
+      })
+    ],
+    behaviours: Behaviour.derive([
+      Focusing.config({}),
+      Keying.config({
+        mode: 'special',
+        onEnter: (comp) => {
+          if (Focus.hasFocus(comp.element)) {
+            Traverse.firstChild(comp.element).each((input) => Focus.focus(input as SugarElement<HTMLElement>));
+            return Optional.some(true);
+          } else {
+            return Optional.none();
+          }
+        },
+        onEscape: goToParent
+      })
+    ])
+  });
+
+  const moveFocus = (parentComp: AlloyComponent, direction: 'left' | 'right'): void => {
+    const nextNode = direction === 'right' ? Traverse.nextSibling : Traverse.prevSibling;
+    const focussed = Arr.find(Traverse.children(parentComp.element), Focus.hasFocus);
+    focussed.fold(Optional.none, nextNode).each((next) => Focus.focus(next as SugarElement<HTMLElement>));
+  };
+
+  const makeStepperButton = (action: VoidFunction, title: string, tooltip: string, classes: string[]) => {
+    const translatedTooltip = backstage.shared.providers.translate(tooltip);
+    return Button.sketch({
+      dom: {
+        tag: 'button',
+        attributes: {
+          'title': translatedTooltip,
+          'aria-label': translatedTooltip
+        },
+        classes: classes.concat(title)
+      },
+      components: [
+        renderIconFromPack(title, backstage.shared.providers.icons)
+      ],
+      action
+    });
+  };
+
+  return {
+    dom: {
+      tag: 'div',
+      classes: [ 'tox-number-input' ]
+    },
+    components: [
+      makeStepperButton(() => decrease(true), 'minus', 'Decrease font size', [ 'highlight-on-focus' ]),
+      memInput.asSpec(),
+      makeStepperButton(() => increase(true), 'plus', 'Increase font size', [ 'highlight-on-focus' ])
+    ],
+    behaviours: Behaviour.derive([
+      Focusing.config({}),
+      Keying.config({
+        mode: 'special',
+        onEnter: (comp) => {
+          if (Focus.hasFocus(comp.element)) {
+            memInput.getOpt(comp).each((inputWrapper) => Focus.focus(inputWrapper.element));
```

Comment:
It might be worth only returning Some if getOpt succeeds.

Reply:
after the refactor to `flow` mode this does not exists anymore